src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -962,14 +962,34 @@ namespace AppInstaller::CLI::Workflow
         bool isUpdate = WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::InstallerExecutionUseUpdate);
 
         IPackageVersion::Metadata installationMetadata;
+        std::vector<Utility::Architecture> requiredArchitectures = Settings::User().Get<Settings::Setting::InstallArchitectureRequirement>();
+        std::vector<Utility::Architecture> optionalArchitectures = Settings::User().Get<Settings::Setting::InstallArchitecturePreference>();
         if (isUpdate)
         {
             installationMetadata = context.Get<Execution::Data::InstalledPackageVersion>()->GetMetadata();
         }
         if (context.Args.Contains(Execution::Args::Type::InstallArchitecture))
         {
+            // arguments override settings.
             context.Add<Execution::Data::AllowedArchitectures>({ Utility::ConvertToArchitectureEnum(std::string(context.Args.GetArg(Execution::Args::Type::InstallArchitecture))) });
         }
+        else 
+        {
+            std::set<Utility::Architecture> settingArchitectures;
+            if (!optionalArchitectures.empty())

```

Comment:
I think the logic here should be:
```
if required exists
    use required
else if preferences exists
    use preferences + Unknown
else
    don't set AllowedArchitectures
```
There shouldn't be a need to combine requirements and preferences.

Reply:
I was thinking that if you wanted only arm or arm64, but of those you may prefer arm64, that may be a reason to have both. But that’s probably taken care of by the existing logic for selecting architectures.

Reply:
The order of the requirements is the order of preference as well, which should take care of that.